src/popup/components/Modals/PayloadForm.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <Modal
+    from-bottom
+    body-without-padding-bottom
+    has-close-button
+    class="payload-form"
+    @close="resolve"
+  >
+    <div class="header text-heading-2">
+      {{ $t('modals.payloadForm.title') }}
+    </div>
+    <div class="description text-description">
+      {{ $t('modals.payloadForm.desc') }}
+    </div>
+    <div class="wrapper">
+      <FormTextarea
+        v-model="value"
+        class="form-textarea"
+        size="sm"
+      >
+        <template #label>
+          {{ $t('modals.payloadForm.label') }}
+        </template>
+      </FormTextarea>
+    </div>
+    <template #footer>
+      <BtnMain
+        class="btn-cancel"
+        variant="secondary"
+        @click="resolve()"
+      >
+        {{ $t('modals.payloadForm.cancelBtn') }}
+      </BtnMain>
+      <BtnMain
+        :disabled="!value.length"
+        @click="submit()"
+      >
+        {{ $t('modals.payloadForm.doneBtn') }}
+      </BtnMain>
+    </template>
+  </Modal>
+</template>
+
+<script lang="ts">
+import { ref, defineComponent, PropType } from '@vue/composition-api';
+import Modal from '../Modal.vue';
+import FormTextarea from '../FormTextarea.vue';
+import BtnMain from '../buttons/BtnMain.vue';
+
+export default defineComponent({
+  name: 'PayloadForm',
+  components: {
+    BtnMain,
+    FormTextarea,
+    Modal,
+  },
+  props: {
+    resolve: { type: Function as PropType<() => void>, required: true },
+    reject: { type: Function as PropType<() => void>, required: true },
+    // eslint-disable-next-line no-unused-vars
+    onChange: { type: Function as PropType<(text: string) => void>, required: true },
```

Comment:
Why do we need `onChange` function? You can return payload value in `resolve` function?